api/v1alpha08/sonataflowplatform_types.go

```diff
@@ -46,6 +46,45 @@ type SonataFlowPlatformSpec struct {
 	// `sonataflow.org/profile: prod`
 	// +optional
 	Services ServicesPlatformSpec `json:"services,omitempty"`
+	// Persistence defines the platform persistence configuration. When this field is set,
+	// the configuration is used as the persistence for platform services and sonataflow instances
+	// that don't provide one of their own.
+	// +optional
+	Persistence *PlatformPersistenceSpec `json:"persistence,omitempty"`
```

Comment:
That was the initial idea, but I found 2 issues with that:

* `PersistencePostgreSql` contains a `JdbcUrl` field, which opens the door for using a database outside the cluster, potentially a security risk. I also don't think it's a good idea to allow the user to enter free text to configure the database here as it's more prone to typos. Let me know if if's there are no such concerns and I will change the code to use the `JdbcUrl` field.
* `PostgreSqlServiceOptions` contains the `databaseSchema`. Each workflow should use its own schema, rather than share it, so to avoid problems that might occur when two workflows interact with the database using the same schema. Instead, the schema used is the name of the workflow itself, thus avoiding any collision that might happen if they all share the same schema.

Reply:
As discussed in slack, the `JdbcUrl` field is required to allow workflows to directly use external DB services without requiring additional configuration in the cluster (via [external name](https://kubernetes.io/docs/concepts/services-networking/service/#externalname) service type in k8s).
For the `databaseSchema`, I will refactor the code so that the contents of `PostgreSqlServiceOptions` minus `databaseSchema` are contained in a different struct, and `PostgreSqlServiceOptions` embeds that struct plus `databaseSchema` so that it's transparent and reused in the current logic.

Reply:
@tchughesiv the embedded struct doesn't work when it comes to JSON un/marshalling. When I run the `make generate-all` command I get this error:
```
sonataflowplatform_services_types.go:93:2: encountered struct field "" without JSON tag in type "PostgreSqlServiceOptions
Error: not all generators ran successfully
```
This is the struct I created:

```
type PostgreSqlServiceOptions struct {
	ServiceOptions
	// Schema of postgresql database to be used.
	// +optional
	DatabaseSchema string `json:"databaseSchema,omitempty"`
}
```

Here is the other structure:

```
type ServiceOptions struct {
	// Name contains the name of the kubernetes service. This field is mandatory.
	// +required
	Name string `json:"name"`
	// Namespace contains the name of the namespace where the kubernetes service resides. This field is optional.
	// +optional
	Namespace string `json:"namespace"`
	// Port contains the port number associated to the kubernetes service. This field is optional.
	// +optional
	Port *int `json:"port,omitempty"`
	// Name of postgresql database to be used. This field is optional.
	// +optional
	DatabaseName string `json:"databaseName,omitempty"`
}
```
Golang accepts the embedded struct and compiles without errors, but when attempting to generate all the files, the `controller-gen` throws this error. 

The only workaround is to change the struct to expose the `ServiceOptions` struct as its own field, but it would then be awkward to have say the name wrapped inside `serviceOption` (`serviceOption.Name`) and `databaseSchema` at a different struct level.
```
type PostgreSqlServiceOptions struct {
	ServiceOptions ServiceOptions `json:"serviceOptions,omitempty"`
	// Schema of postgresql database to be used.
	// +optional
	DatabaseSchema string `json:"databaseSchema,omitempty"`
}
```

I can only think of having each unique struct for both `WorkFlow` and `Platform` as I proposed in this PR, even if that means repeating fields. WDYT?

Reply:
> @tchughesiv the embedded struct doesn't work when it comes to JSON un/marshalling. When I run the `make generate-all` command I get this error:
> 
> ```
> sonataflowplatform_services_types.go:93:2: encountered struct field "" without JSON tag in type "PostgreSqlServiceOptions
> Error: not all generators ran successfully
> ```
> 
> This is the struct I created:
> 
> ```
> type PostgreSqlServiceOptions struct {
> 	ServiceOptions
> 	// Schema of postgresql database to be used.
> 	// +optional
> 	DatabaseSchema string `json:"databaseSchema,omitempty"`
> }
> ```

@jordigilh adding `json:",inline"` to the embedded structs should do the trick. for example:
```golang
type PostgreSqlServiceOptions struct {
	ServiceOptions `json:",inline"`
	// Schema of postgresql database to be used.
	// +optional
	DatabaseSchema string `json:"databaseSchema,omitempty"`
}
```

Reply:
Yeah, it worked. Thanks!!!